### PR TITLE
feat(CodeStream): New draft content

### PR DESCRIPTION
**Pull request by paperclypse**
New CodeStream draft content.

Hold for release April 7ish.

### CODE REVIEW

File: src/content/docs/codestream/how-use-codestream/performance-monitoring.mdx [hunk 1]
```diff
@@ -19,6 +19,66 @@ Go here to get or create your [New Relic user key](https://one.newrelic.com/laun
 
 Once you have your user key, in **Observability** click **Connect to New Relic One**, then paste your user key and click **Connect**.
 
+## Monitor code-level metrics [#code-level]
+
+CodeStream's code-level metrics give you detailed insight into how your code is performing.
+
+![A screenshot of a Python method with average duration, throughput, and error rate telemetry.](./images/code-level-metrics.png "A Python method with average duration, throughput, and error rate metrics.")
+
+<figcaption>When relevant, CodeStream's code-level metrics are shown in a CodeLens above your Python methods.</figcaption>
```
Comment (bradleycamacho) [inline src/content/docs/codestream/how-use-codestream/performance-monitoring.mdx:28]: It doesn't seem like we define `CodeLens` anywhere in this doc, or link out to it. Could probably just add a link to wherever it is defined!

File: src/content/docs/codestream/how-use-codestream/performance-monitoring.mdx [hunk 2]
```diff
@@ -19,6 +19,66 @@ Go here to get or create your [New Relic user key](https://one.newrelic.com/laun
...
+
+<Callout variant="caution">
```
Comment (bradleycamacho) [inline src/content/docs/codestream/how-use-codestream/performance-monitoring.mdx:30]: This is fine, but I wonder if we can avoid using a callout here by changeing the title of the section to something like `Montitor code-level metrics (Python only)`

Your call on this one!

File: src/content/docs/codestream/how-use-codestream/performance-monitoring.mdx [hunk 3]
```diff
@@ -19,6 +19,66 @@ Go here to get or create your [New Relic user key](https://one.newrelic.com/laun
...
+Code-level metrics are only available for Python code and [supported Python frameworks](/docs/apm/agents/python-agent/getting-started/instrumented-python-packages). These metrics are shown via CodeLens text, a line of text inserted above your instrumented method.
+
+Your Python application code also needs to be monitored by the Python APM agent and your repository associated with a New Relic service.
```
Comment (bradleycamacho) [inline src/content/docs/codestream/how-use-codestream/performance-monitoring.mdx:33]: There could be value in adding a link directly to the python agent here. I'm not entirely sure who the audience is here, but if it includes prospective NR users who currently only use codestream, a link to the python intro doc could onboard some people!

File: src/content/docs/codestream/how-use-codestream/performance-monitoring.mdx [hunk 4]
```diff
@@ -19,6 +19,66 @@ Go here to get or create your [New Relic user key](https://one.newrelic.com/laun
...
+</Callout>
+
+Click the CodeLens text to see related charts.
+
+![A screenshot of an expanded code-level metrics CodeLens that shows related charts.](./images/code-level-charts.png "An expanded code-level metrics CodeLens that shows related charts.")
+
+<figcaption>Click the entity name at the top of the charts to jump to this data in New Relic One.</figcaption>
+
+Code-level metrics focus on your "golden metrics" for the last 30 minutes. These metrics are: latency, traffic, errors, and saturation.
```
Comment (bradleycamacho) [inline src/content/docs/codestream/how-use-codestream/performance-monitoring.mdx:42]: I seem to be very pro-links today. Maybe we should link out to a doc that defines `golden metrics` as a CodeStream user new to observability may not know what that means.

Comment (bradleycamacho) [reply src/content/docs/codestream/how-use-codestream/performance-monitoring.mdx:42]: Scratch that, I should have read the second sentence 😅

File: src/content/docs/codestream/how-use-codestream/performance-monitoring.mdx [hunk 5]
```diff
@@ -19,6 +19,66 @@ Go here to get or create your [New Relic user key](https://one.newrelic.com/laun
...
+The related charts include some standard fields:
+
+* The name of the method being monitored. 
+* **Entity**: A dropdown that switches between what's being monitored, for example methods in development or production repositories. 
+* **Repo**: Your current repository.
+* **File**: The file that contains the method.
+
+### Enable code-level metrics [#enable]
+
+If you've enabled CodeStream and are monitoring entities in New Relic, but you haven't associated your current repository with New Relic, Visual Studio Code shows a CodeLens at the top of your open file.
```
Comment (bradleycamacho) [inline src/content/docs/codestream/how-use-codestream/performance-monitoring.mdx:53]: Is this only possible for VSCode? I know CodeStream also supports other IDEs.

File: src/content/docs/codestream/how-use-codestream/performance-monitoring.mdx [hunk 6]
```diff
@@ -19,6 +19,66 @@ Go here to get or create your [New Relic user key](https://one.newrelic.com/laun
...
+
+If you don't want to be prompted for this in future, click the **Don't show** checkbox.
+
+### Change your CodeLens format  [#format]
+
+You can use your IDE's settings to change how your CodeLens messages are shown: 
+
+* In VS Code, go to Settings, and search for "codestream".
+* In JetBrains, go to **Preferences/Settings > Tools > CodeStream**
+* In Visual Studio, go to **Tools > Options > CodeStream**.
```
Comment (bradleycamacho) [inline src/content/docs/codestream/how-use-codestream/performance-monitoring.mdx:69]: Not part of the review, but I had no idea Visual Studio was still a supported IDE!

**Review by bradleycamacho — DISMISSED** [review]
Looks really great! Left a few comments.

File: src/content/docs/codestream/how-use-codestream/performance-monitoring.mdx (hunk 1 above)
Comment (paperclypse) [reply src/content/docs/codestream/how-use-codestream/performance-monitoring.mdx:50]: I rewrote this sentence slightly to clarify.

File: src/content/docs/codestream/how-use-codestream/performance-monitoring.mdx (hunk 2 above)
Comment (paperclypse) [reply src/content/docs/codestream/how-use-codestream/performance-monitoring.mdx:52]: This is a good suggestion and I wish I didn't feel the need to leave this caution here. Other languages will be added eventually.

File: src/content/docs/codestream/how-use-codestream/performance-monitoring.mdx (hunk 3 above)
Comment (paperclypse) [reply src/content/docs/codestream/how-use-codestream/performance-monitoring.mdx:55]: Done!

File: src/content/docs/codestream/how-use-codestream/performance-monitoring.mdx (hunk 5 above)
Comment (paperclypse) [reply src/content/docs/codestream/how-use-codestream/performance-monitoring.mdx:75]: I've revised slightly to remove the VS Code mention

File: src/content/docs/codestream/how-use-codestream/performance-monitoring.mdx (hunk 6 above)
Comment (paperclypse) [reply src/content/docs/codestream/how-use-codestream/performance-monitoring.mdx:91]: Apparently so!

**Comment by paperclypse** [comment]
HOLD until April 6 or 7

**Comment by paperclypse** [comment]
got my review, merging it in

**Pull request merged**